### ISSUE TITLE
Building 3.4-stable tag warning C4346: 'BVH_ABB<Bounds,Point>::ConvexHull': dependent name is not a type

### ISSUE THREAD

**Issue by Giwayume** [godotengine/godot#55069]
### Godot version

3.4 stable

### System information

Windows 7 64 bit

### Issue description

Has something significant changed about the build dependencies from 3.3 to 3.4? Getting this error.

```
$ scons p=windows target=release_debug tools=yes module_mono_enabled=no
scons: Reading SConscript files ...
Configuring for Windows: target=release_debug, bits=default
Found MSVC version 14.0, arch amd64, bits=64
Checking for C header file mntent.h... (cached) no
scons: done reading SConscript files.
scons: Building targets ...
Compiling ==> platform\windows\godot_windows.cpp
godot_windows.cpp
Compiling ==> platform\windows\crash_handler_windows.cpp
crash_handler_windows.cpp
C:\Program Files (x86)\Windows Kits\8.1\include\um\imagehlp.h(1873): warning C4091: 'typedef ': ignored on left of '' when no variable is declared
C:\Program Files (x86)\Windows Kits\8.1\include\um\imagehlp.h(3519): warning C4091: 'typedef ': ignored on left of '' when no variable is declared
Compiling ==> platform\windows\os_windows.cpp
os_windows.cpp
Z:\GitHub\godot\core/math/bvh_abb.h(107): warning C4346: 'BVH_ABB<Bounds,Point>::ConvexHull': dependent name is not a type
Z:\GitHub\godot\core/math/bvh_abb.h(107): note: prefix with 'typename' to indicate a type
Z:\GitHub\godot\core/math/bvh_abb.h(274): note: see reference to class template instantiation 'BVH_ABB<Bounds,Point>' being compiled
Z:\GitHub\godot\core/math/bvh_abb.h(107): error C2061: syntax error: identifier 'ConvexHull'
scons: *** [platform\windows\os_windows.windows.opt.tools.64.obj] Error 2
scons: building terminated because of errors.
[Time elapsed: 00:00:08.846]
```

This is the latest C++ installed

![image](https://user-images.githubusercontent.com/4075314/142286963-0fec7542-d86e-4095-9251-66dbd20d899f.png)

Python versions

![image](https://user-images.githubusercontent.com/4075314/142287009-dbfdcb20-d5ea-49c6-b0cd-5c653fcfbdc2.png)

scons

![image](https://user-images.githubusercontent.com/4075314/142287290-854d489b-ed1d-4c11-8ff2-0c904846bcf7.png)

Visual studio build tools

![image](https://user-images.githubusercontent.com/4075314/142288249-2de5bdde-abf3-4bfc-a0ba-684bf009d7f4.png)

### Steps to reproduce

Not sure if this is reproducible on other systems or just me...

**Comment by Giwayume** [comment]
Changing this line to add `typename` allows it to compile for me.

https://github.com/godotengine/godot/blob/3.4-stable/core/math/bvh_abb.h#L107

`uint32_t find_cutting_planes(const typename BVH_ABB::ConvexHull &p_hull, uint32_t *p_plane_ids) const {`

Has no one else run into this issue?

**Comment by lawnjelly** [comment]
This is down to the zealousness of the particular compiler I think. There are no problems in the CI compiler and clang, but we can add this if it flags as a warning in a particular version of microsoft compiler.

**Comment by akien-mga** [comment]
Fixed by #55079.
